lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -592,6 +593,9 @@ sub failed_modules_with_needles {
             for my $needle (@{$detail->{needles}}) {
                 push( @needles, [ $needle->{name}, $counter ] );
             }
+            if (!@{$detail->{needles}}) {
+                push(@needles, [ undef, $counter ] );
```

Comment:
that might be your personal preference, but it looks to me this is not openQA perl style:

<pre>
coolo@gertrude#openQA>grep -r 'if.*([^ ]' lib/ | wc -l
405
coolo@gertrude#openQA>grep -r 'if.*( ' lib/ | wc -l
22
</pre>